src/frequenz/sdk/timeseries/formula_engine/_formula_steps.py

```diff
@@ -390,7 +421,37 @@ async def fetch_next(self) -> Sample[QuantityT] | None:
         Returns:
             The fetched Sample.
         """
-        self._next_value = await self._stream.receive()
+        if self._fallback is not None and self._fallback.is_running:
+            tasks = [self._stream.receive(), self._fallback.fetch_next()]
+            result = await asyncio.gather(*tasks, return_exceptions=False)
+
+            # Synchronize the timestamps from two sources
+            while result[0].timestamp < result[1].timestamp:
+                result[0] = await self._stream.receive()
+
+            while result[0].timestamp > result[1].timestamp:
+                result[1] = await self._fallback.fetch_next()
```

Comment:
I feel like the first check shouldn't be there.  The fallback should be updated to match the original stream, not the other way around.  Else we could skip data.

If the fallback stream is ahead, then we shouldn't fetch from it until the original stream has caught up, and send the values from the original stream until then.

I guess this would also mean that you can't pull from both the streams together using a `asyncio.gather` either.

Reply:
Done